pkg/logs/decoder/handler.go

```diff
@@ -0,0 +1,160 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package decoder
+
+import "github.com/DataDog/datadog-agent/pkg/logs/parser"
+
+// Handler should replace LineHandler.
+// Handler defines the methods for handling Lines and form a Output ready Line.
+type Handler interface {
+	Handle(line *RichLine)
+	SendResult()
+	Cleanup()
+}
+
+// SingleHandler treats the incoming Line independently and prepare the results
+// for send.
+type SingleHandler struct {
+	truncator LineTruncator
+}
+
+// NewSingleHandler creates a new instance of SingleHandler.
+func NewSingleHandler(truncator LineTruncator) *SingleHandler {
+	return &SingleHandler{
+		truncator: truncator,
+	}
+}
+
+// Handle takes a line, truncates accordingly.
+func (s *SingleHandler) Handle(line *RichLine) {
+	s.truncator.truncate(line)
+}
+
+// Cleanup closes the downstream operations.
+func (s *SingleHandler) Cleanup() {
+	s.truncator.Close()
+}
+
+// SendResult does nothing since single line handler doesn't cache the lines.
+func (s *SingleHandler) SendResult() {}
+
+// LineTruncator truncates a large line into multiple small lines with
+// shared Prefix. The results are then sent to outputChan for further
+// operations.
+type LineTruncator struct {
+	outputChan chan *Output
+	maxLen     int // max send length
+}
+
+// NewLineTruncator creates a new instance of LineTruncator.
+func NewLineTruncator(outputChan chan *Output, maxLen int) *LineTruncator {
+	return &LineTruncator{
+		outputChan: outputChan,
+		maxLen:     maxLen,
+	}
+}
+
+// Close closes the output channel. This method should be called by the sender of
+// outputChan, normally the Cleanup method of upstream.
+func (l *LineTruncator) Close() {
+	close(l.outputChan)
+}
+
+// truncate splits the large line to multiple smaller size lines with the same prefix
+// and send them to outputChan.
+func (l *LineTruncator) truncate(line *RichLine) {
+	if l.isLarge(line) {
+		lines := l.split(line)
+		for _, ln := range lines {
```

Comment:
We use full names in most of the logs code base ...
```suggestion
		for _, line := range lines {
```

Reply:
Agreed, it's a valid remark. 
Here `line` refers to RichLine, and everywhere in the code, when passed as an argument, we call RichLine `line`. So this is more about keeping the code consistent.